src/dbnode/storage/cleanup.go

```diff
@@ -185,6 +175,41 @@ func (m *cleanupManager) Cleanup(t time.Time, isBootstrapped bool) error {
 	return multiErr.FinalError()
 }
 
+func (m *cleanupManager) ColdFlushCleanup(t time.Time, isBootstrapped bool) error {
+	// Don't perform any cleanup if we are not boostrapped yet.
+	if !isBootstrapped {
+		m.logger.Debug("database is still bootstrapping, terminating cleanup")
+		return nil
+	}
+
+	m.Lock()
+	m.cleanupInProgress = true
```

Comment:
Can't WarmFlushCleanup and ColdFlushCleanup be called at the same time now? Perhaps need a new variable to distinguish between "warmFlushCleanupInProgress" and "coldFlushCleanupInProgress"?

Reply:
Or are they used at different times?

Reply:
There was no prior protection for concurrent calls since it gets called in a single thread. Now that the calls are split they each run in their own thread. I don't think they need protection against being run concurrently as they don't conflict. 

I did update the state tracking to properly report on the cold/warm flush cleanup in progress metric.

Reply:
That makes sense, I only later realized they are used from two different callsites 👍